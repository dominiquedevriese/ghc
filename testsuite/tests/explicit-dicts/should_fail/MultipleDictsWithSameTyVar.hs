module MultipleDictsWithSameTyVar where

import GHC.Classes (Eq(..))
import GHC.Show (Show(..))

f :: (Eq a, Show a) => a -> a
f = undefined

g :: Eq.Dict Bool -> Show.Dict Bool -> Bool -> Bool
g eqBoolDict showBoolDict = f ((eqBoolDict :: Eq a)) ((showBoolDict :: Show a))
