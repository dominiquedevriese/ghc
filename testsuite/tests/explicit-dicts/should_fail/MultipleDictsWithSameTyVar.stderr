
MultipleDictsWithSameTyVar.hs:10:29: error:
    • Constraint ‘Show a’ not found in:
        Show Bool => Bool -> Bool
    • In the expression:
        f ((eqBoolDict :: Eq a)) ((showBoolDict :: Show a))
      In an equation for ‘g’:
          g eqBoolDict showBoolDict
            = f ((eqBoolDict :: Eq a)) ((showBoolDict :: Show a))
